src/routes/contractViewState.js

```diff
@@ -0,0 +1,52 @@
+const warp = require('../warp');
+const { config } = require('../config');
+const { getLastStateFromDreCache, getCachedViewState, insertViewStateIntoCache } = require('../db/nodeDb');
+
+module.exports = async (ctx) => {
+  if (!config.availableFunctions.viewState) {
+    ctx.body = 'Contract view state functionality is disabled';
+    ctx.status = 503;
+    return;
+  }
+
+  const nodeDb = ctx.nodeDb;
+
+  try {
+    const contractId = ctx.query.id;
+    if (!contractId) {
+      throw new Error('Missing contract id');
+    }
+    if (!isTxIdValid(contractId)) {
+      throw new Error('Invalid tx format');
+    }
+    if (!ctx.query.input) {
+      throw new Error('Missing input');
+    }
+    const input = JSON.parse(decodeURIComponent(ctx.query.input));
+
+    let interactionResult = null;
+    const lastDreState = await getLastStateFromDreCache(nodeDb, contractId);
+    let sortKey = lastDreState.sort_key;
+    let cachedView = (await getCachedViewState(nodeDb, contractId, sortKey, JSON.stringify(input)))[0];
+
+    if (cachedView) {
+      interactionResult = JSON.parse(cachedView.result);
+    } else {
+      interactionResult = await warp.contract(contractId).viewState(input);
+      sortKey = (await warp.stateEvaluator.latestAvailableState(contractId)).sortKey;
+
+      cachedView = await insertViewStateIntoCache(nodeDb, contractId, sortKey, input, interactionResult);
+    }
+
+    ctx.body = { viewState: interactionResult, sortKey, signature: cachedView.signature, hash: cachedView.view_hash };
+    ctx.status = 200;
+  } catch (e) {
+    ctx.body = e.message;
+    ctx.status = 500;
+  }
+};
+
+function isTxIdValid(txId) {
```

Comment:
can we just export it from the sdk? i even used it in warpik :)

Reply:
For now, I moved this function to a seperate file: common.js so we can use one implementation.
I didn't change it everywhere, because I was testing only the viewState functionality. 
isTxIdValid in SDK ++